enterprise/server/scheduling/priority_task_scheduler/priority_task_scheduler.go

```diff
@@ -474,6 +478,60 @@ func (q *PriorityTaskScheduler) canFitTask(res *scpb.EnqueueTaskReservationReque
 	return true
 }
 
+// This function peeks at the front of the queue and checks to see if the task
+// still exists--if it doesn't, then we know the task was picked up *and
+// finished* by another worker, so we can safely prune it from the local queue,
+// allowing our queue size to go down even when we aren't picking up new tasks.
+//
+// During sudden bursts of large tasks, the task queues on individual executors
+// can get pretty long.  If we are autoscaling according to task queue length,
+// then doubling the number of executors might completely empty out the "real"
+// task queue, but if the original executor pool is fully stuck on large tasks,
+// the average task queue length will only be cut in half!  If that new length
+// is still over our autoscaling threshold, we'll scale up more even though we
+// don't need to.  Trimming makes this case less likely.
+func (q *PriorityTaskScheduler) trimQueue() {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+
+	if q.shuttingDown || q.q.Len() == 0 {
+		return
+	}
+	nextTask := q.q.Peek()
+	if nextTask == nil {
+		return
+	}
+	ctx := log.EnrichContext(q.rootContext, log.ExecutionIDKey, nextTask.GetTaskId())
+	ctx, cancel := context.WithCancel(ctx)
+	ctx = tracing.ExtractProtoTraceMetadata(ctx, nextTask.GetTraceMetadata())
+
+	go func() {
+		defer cancel()
```

Comment:
hm this comment somehow didn't make it through -

I think the ctx cancelation is not needed here since the goroutine that does the cancelation isn't spawning any background goroutines that would need to be canceled?

Reply:
Ah sure, especially moot now that the goroutine is gone.